servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ReplayStrategies.java

```diff
@@ -158,23 +157,29 @@ private static final class LazyTimeLimitedReplayAccumulator<T> implements Replay
 
         @Override
         public void accumulate(@Nullable final T t) {
+            final long nanoTime = executor.currentTime(NANOSECONDS);
+            trimExpired(nanoTime);
             if (items.size() >= maxItems) {
                 items.poll();
             }
-            items.add(new TimeStampSignal<>(executor.currentTime(NANOSECONDS), t));
+            items.add(new TimeStampSignal<>(nanoTime, t));
         }
 
         @Override
         public void deliverAccumulation(final Consumer<T> consumer) {
-            final Iterator<TimeStampSignal<T>> itr = items.iterator();
-            final long nanoTime = executor.currentTime(NANOSECONDS);
-            while (itr.hasNext()) {
-                final TimeStampSignal<T> next = itr.next();
-                if (nanoTime - next.timeStamp >= ttlNanos) {
-                    itr.remove();
-                } else {
-                    consumer.accept(next.signal);
-                }
+            if (items.isEmpty()) {
+                return;
+            }
+            trimExpired(executor.currentTime(NANOSECONDS));
```

Comment:
this requires multiple iterations of the collection, can we keep a single iteration?